alibi_detect/saving/saving.py

```diff
@@ -203,12 +211,19 @@ def write_config(cfg: dict, filepath: Union[str, os.PathLike]):
     filepath
         Filepath to directory to save 'config.toml' file in.
     """
+    # Create directory if it doesn't exist
     filepath = Path(filepath)
     if not filepath.is_dir():
         logger.warning('Directory {} does not exist and is now created.'.format(filepath))
         filepath.mkdir(parents=True, exist_ok=True)
+    # Convert pathlib.Path's to str's
     cfg = _path2str(cfg)
+    # Validatre config before final tweaks
+    validate_config(cfg)  # Must validate here as replacing None w/ str will break validation
+    # Replace None with "None", and dicts with integer keys with str keys (TODO: S2C depending on toml library updates)
```

Comment:
What's `S2C`?

Reply:
Sorry, subject to change. I've written it out properly now!